cli/src/status.rs

```diff
@@ -0,0 +1,97 @@
+use anyhow::Result;
+use kube::{api::ListParams, Api, Client, ResourceExt};
+use linkerd_failover_controller::TrafficSplit;
+use serde::Serialize;
+use std::fmt::Display;
+
+#[derive(Serialize)]
+pub struct TrafficSplitStatus {
+    namespace: String,
+    name: String,
+    status: FailoverStatus,
+    services: Vec<String>,
+}
+
+#[derive(Serialize)]
+#[serde(rename_all = "lowercase")]
+enum FailoverStatus {
+    Primary,
+    Fallback,
+}
+
+impl Display for FailoverStatus {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match *self {
+            Self::Primary => std::fmt::Display::fmt("Primary", f)?,
+            Self::Fallback => std::fmt::Display::fmt("Fallback", f)?,
+        };
+        Ok(())
```

Comment:
nit: can be a bit tighter:

```suggestion
        match *self {
            Self::Primary => f.write_str("Primary"),
            Self::Fallback => f.write_str("Fallback"),
        }
```

Reply:
actually, `write_str` ignores any formatting options on the `Formatter` so you lose e.g. width padding unless you use `fmt`.

Reply:
Ah! In that case `f.pad("Primary")` could work https://doc.rust-lang.org/std/fmt/struct.Formatter.html#method.pad